x/validators/init.go

```diff
@@ -15,7 +15,7 @@ var _ weave.Initializer = Initializer{}
 
 // FromGenesis will parse initial account info from genesis
 // and save it to the database
-func (Initializer) FromGenesis(opts weave.Options, kv weave.KVStore) error {
+func (Initializer) FromGenesis(opts weave.Options, params weave.GenesisParams, kv weave.KVStore) error {
```

Comment:
This would actually be a case to use it.
Right now, we blindly support changes.

We could store the current validator set in the kvstore, update on every succesful tx. And then update the handler to reject tx that are redundant (eg. reducing non-existant validator power to 0, setting an existant validator power to current value).

This would at least make use of this feature and demo it is working. Especially, if you keep that code loosly coupled to the rest of the package, it would be an example for using this field that could be used in other packages.

Reply:
Yes, that makes sense now.

Reply:
Actually, and maybe out of scope of this PR, the app.{Base,Store}App could save this info and pass it in the header on every request. All changes to the validator set and collected in BaseApp.DeliverTx and applied on BaseApp.EndBlock, so we have access to everything there to maintain a proper set (both the set of validators for the last block - to reference the commit signatures, as well as the current set after applying all transactions in this block).

I think just doing some accounting in the x/validators package is a nice step forward. When validated, it can be pulled out to app-level.

I would not use a bucket, but just a singleton. Kind of like we use for _c:cash Configuration. But you don't need to use a framework. Just a protobuf message to encode, and pick a unique key to write directly to the kvstore